ksmjwt_localstack_test.go

```diff
@@ -0,0 +1,101 @@
+package kmsjwt_test
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/kms/types"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/spacelift-io/kmsjwt/v6"
+)
+
+func TestWithLocalStack(t *testing.T) {
+	const in = "sign me, please"
+	ctx := context.Background()
+	client := newClient(t, ctx)
+	keyID := client.CreateKey(t, ctx)
+	publicKey := client.GetPublicKey(t, ctx, keyID)
+
+	t.Run("new", func(t *testing.T) {
+		signer := kmsjwt.New(client.KMS, keyID)
+
+		signature, err := signer.Sign(in, ctx)
+		require.NoError(t, err, "sign")
+
+		err = signer.Verify(in, signature, ctx)
+		assert.NoError(t, err, "verify")
+	})
+
+	t.Run("new with public key", func(t *testing.T) {
+		signer := kmsjwt.NewWithPublicKey(client.KMS, keyID, publicKey)
+
+		signature, err := signer.Sign(in, ctx)
+		require.NoError(t, err, "sign")
+
+		err = signer.Verify(in, signature, ctx)
+		assert.NoError(t, err, "verify")
+	})
+
+	t.Run("RFC compliance", func(t *testing.T) {
+		signer := kmsjwt.New(client.KMS, keyID)
+
+		signature, err := signer.Sign(in, ctx)
+		require.NoError(t, err, "sign")
+
+		builtinSigner := jwt.GetSigningMethod(signer.Alg())
+		require.NotNil(t, builtinSigner, "unknown algorithm")
+
+		err = builtinSigner.Verify(in, signature, publicKey)
+		assert.NoError(t, err, "verify")
+	})
+}
+
+func newClient(t *testing.T, ctx context.Context) Client {
+	t.Helper()
+
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithRegion("eu-west-1"),
+		config.WithBaseEndpoint("http://localhost:4566"),
```

Comment:
A bit of a nit. That would make sense to be able to configure the test endpoint.

Reply:
Yeah. I wanted to move quick and dirty without adding options (thoughtfully). Like, ideally it would be easy to just run against a real key in real AWS... to spare time I just went with this dirty option. If someone needs it of course they can enhance. So I won't do this.

(I actually did run it against real KMS key, too.)

Reply:
Make sense 👍🏻 

Maybe we can just drop a few instructions on the readme about that? Like explain in 1 line that running tests depends on a local localstack running. And also explain that it's maybe also worth commenting out that to run that against a real KMS if we want to be 100% sure a change will not break.